editor/static/js/question/edit.js

```diff
@@ -771,20 +771,29 @@ $(document).ready(function() {
 
             //  Finally, mark duplicate names
             names.sort(Numbas.util.sortBy('name'));
+            var groupArray = [];
             function handle_group(group) {
-                group.forEach(function(n) {
-                    n.v.duplicateNameError(group.length > 1 ? n.name : null);
+                group.forEach(function (n) {
+                    n.v.duplicateNameError(null);
```

Comment:
While it won't noticeably affect performance, you can avoid having two nested `forEach` calls by setting `n.v.duplicateNameError(null)` when you loop over each name, instead of here where you're looping over names inside groups.